stdlib/seq.mc

```diff
@@ -191,12 +195,14 @@ utest filter (lam x. eqi x 1) [1,2,4] with [1]
 utest filter (lam. false) [3,5,234,1,43] with [] using eqSeq eqi
 utest filter (lam x. gti x 2) [3,5,234,1,43] with [3,5,234,43]
 
-recursive
-  let find = lam p. lam seq.
-    if null seq then None ()
-    else if p (head seq) then Some (head seq)
-    else find p (tail seq)
-end
+let find = lam p. lam seq.
+  foldl
+    (lam acc. lam x.
+      match acc with Some _ then acc
+      else if p x then Some x
+      else None ())
+    (None ())
+    seq
```

Comment:
The old version can exit early while the new version needs to go through the entire sequence. But I'm guessing the new version is cheaper in many cases because of `foldl`?

Reply:
Yes. At least in `boot` it was a few seconds faster. I'm not sure about the compiled version though (it runs so fast that it's hard to distinguish improvements from runtime variances). We should probably have benchmarks specifically for functions like this.